signac/__main__.py

```diff
@@ -409,13 +410,7 @@ def _sig(st):
         destination = get_project(root=args.destination)
     except LookupError:
         if args.allow_workspace:
-            destination = Project(
-                config={
-                    "project": os.path.relpath(args.destination),
-                    "project_dir": args.destination,
-                    "workspace_dir": ".",
-                }
-            )
+            destination = Project(os.path.relpath(args.destination))
```

Comment:
Good point, I obviously didn't really think this through. I'm not sure what to do with this. In the case where the workspace is a direct subdirectory of the project we could still support this (for the lifetime of 1.x) by doing
```
try:
    destination = Project(os.path.dirname(os.path.relpath(args.destination)))
except LookupError:
    ...
```
Is that worthwhile to avoid a breaking change here?

Reply:
I personally think the “allow workspace” feature is a little unintuitive (or unexpected?) when everything else in signac is based on a project. There are no other functions I know of that deal with raw workspace directories. I’d be fine with removing it. I think @csadorf might have added this feature, so maybe helpful to get another opinion on whether we need it in 2.0.

Reply:
As for the lifetime of 1.x, this branch targets 2.0, right? I don’t see the connection. A breaking change is allowable.

Reply:
It's been a couple days since I looked at this but I hadn't finished my review. I think the best action to take here is to remove `--allow-workspace` in a separate PR. We could do that first, or we can mark this as TODO and leave it partially broken with an issue to fix it later.

Reply:
> As for the lifetime of 1.x, this branch targets 2.0, right? I don’t see the connection. A breaking change is allowable.
Yeah sorry I was context switching between too many PRs simultaneously when I wrote that. You're right, we can break this. I'll leave a TODO item for a future PR.